### PR TITLE
Diamond Strategy added

### PR DESCRIPTION
**Pull request by mablue**
It is AfghanWoman Strategy.
That takes her own rights like Afghanstan women
Those who still proud and hopeful.
Those who the most beautiful creatures in the depths of the darkest.
Those who shine like diamonds buried in the heart of the desert ...
Why not help when we can?
If we believe there is no man left with them
(Which is probably the product of the thought of painless corpses)
Where has our humanity gone?
Where has humanity gone?
Why not help when we can?
24Chira-superJumbo

### CODE REVIEW

**Review by xmatthias — COMMENTED** [review]
Maybe i'm missing something - but this strategy seems almost identical to MultiMa.

Unless it has some distinct differences (which doesn't make it a copy with 3 lines difference) - i don't think it makes sense to include it here.
While more selection is usually good - having 100 almost identical strategies will not make sense.

**Comment by mablue** [comment]
yes it have 3 key change 👍 
plus and minus DMs cross added
dualfit added 
sma main indicator changed to ema, 
but maybe yes the main idea still not changed.
i commited this cuz of that it have some benefits whit this changes...
 I decided to keep multima woman still nake
and af wear hijab(limit it with some other conditions) to multima as an other strategy that named AfghanWoman.
It make less but more effective signals than multima
like a dimond that shine better than other materials but you cant see it like gold and silver, everywhere 😎😂

**Comment by xmatthias** [comment]
> I decided to keep multima woman still nake

Please watch your language. 
This is seen as offensive - and this is exactly the point i was trying to make above.

As such - please change the name of the strategy, otherwise i'll for sure be able to accept this.

**Comment by mablue** [comment]
> > I decided to keep multima woman still nake
> 
> Please watch your language.
> This is seen as offensive - and this is exactly the point i was trying to make above.
> 
> As such - please change the name of the strategy, otherwise i'll for sure be able to accept this.

ok I will change it and commit again :) thanks sorry if I dont know the exact meaning of words...I mean free!

File: user_data/strategies/Dimond.py [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+# 𝒲𝒽𝓎 𝓃𝑜𝓉 𝒽𝑒𝓁𝓅 𝓌𝒽𝑒𝓃 𝓌𝑒 𝒸𝒶𝓃?
+# IMPORTANT: This strategy
+# designed for "ZERO" loss and "UNDER"
+# 15 minuts avg duration.So if you have more
+# loss and more avg, Its "NOT" normal result, and
+# you will change config.json variables and hyperoption commands
+# Thanks To @xmatthias if he was approve the last version of This strategy
+# That just a lazy code. I never can reach to this strategy(Now its really a dimond.)
+# * freqtrade hyperopt --hyperopt-loss ShortTradeDurHyperOptLoss --spaces all --strategy 𝒟𝒾𝓂𝑜𝓃𝒹 -e 700 -j 2 --timerange 20210810-20210813
+# * freqtrade backtesting --strategy 𝒟𝒾𝓂𝑜𝓃𝒹
```
Comment (xmatthias) [inline user_data/strategies/Dimond.py:20]: I don't mind providing sample code for hyperopting (that can prove to be quite helpful) - but then it should be working code (which means - standard characters).

**Review by xmatthias — APPROVED** [review]

**Comment by Kroissan** [comment]
Hello, thanks mablue for everything u made. I get incredible result backtesting your new strategy. However when i switch to production mode, no trade is made :/
Logs are happy just sending heartbeat. 
I use docker:latest with ta lib installed (also tried with :stable). It work when backtesting but not in production mode.

**Comment by mablue** [comment]
> Hello, thanks mablue for everything u made. I get incredible result backtesting your new strategy. However when i switch to production mode, no trade is made :/
> Logs are happy just sending heartbeat.
> I use docker:latest with ta lib installed (also tried with :stable). It work when backtesting but not in production mode.

thanks. yes maybe I didnt checked protection mode befour The idea is that, not using any protection will be better than a beta protection for now.

**Comment by Kroissan** [comment]
thanks. yes maybe I didnt checked protection mode befour The idea is that, not using any protection will be better than a beta protection for now.

**Comment by Kroissan** [comment]
I mean just the normal trading mode in real time it doesn't do anything

**Comment by mablue** [comment]
> I mean just the normal trading mode in real time it doesn't do anything

@xmatthias  what is the problem? 😅 why some times backtest is very good but nothing happen in real?I think cuz of less signals ye?!

**Comment by Kroissan** [comment]
Strange thing is. I did a backtest of the last 6h and the backtest results did some trade. When the bot was running irl on the same period it didn't buy... Are you using the docker image?

**Comment by mablue** [comment]
> Strange thing is. I did a backtest of the last 6h and the backtest results did some trade. When the bot was running irl on the same period it didn't buy... Are you using the docker image?

no im installed from scratch

**Comment by drakes00** [comment]
Btw, the file is named Dimond.py and not Diamond.py ;-)

**Comment by drakes00** [comment]
If you look at the hyperopt results, only one trade was made: 

    # *    6/700:      1 trades. 1/0/0 Wins/Draws/Losses. Avg profit  17.68%. Median profit  17.68%. Total profit  58.94100000 USDT (   5.89Σ%). Avg duration 0:00:00 min. Objective: 1.79949

I don't know what was the timerange but it seems a bit conservative on the buy signals and would explain why it's not doing anything.
If I backtest on July, I don't have any trade at all with default parameters.

**Comment by xmatthias** [comment]
Hyperopting on the data / pairs you're going to use it on is an important point for sure.

**Comment by Kroissan** [comment]
image
I hyperopt it with my config and pairs (timerange from 20210815- it did approx 100 trades). It would have done 13 trades today i tried everything (reinstall from scratch and so on) i want to tear my hair out.

**Comment by drakes00** [comment]
You can also export and plot the data to check when the trades would have done.
Maybe, all backtested trades have been done prior in the day to when you launched the bot in prod?
(No trade made since 11:30AM when I switched strategy to test but buy signals shown in UI in the morning)

**Comment by mablue** [comment]
> image
> I hyperopt it with my config and pairs (timerange from 20210815- it did approx 100 trades). It would have done 13 trades today i tried everything (reinstall from scratch and so on) i want to tear my hair out.

it cuz of that rejected other buy signals you can see for one day (4 min) its real real good profit (15%) :))) what you want. change to usdt maybe it work

**Comment by drakes00** [comment]
Actually, it seems that I can confirm.
Buy signals showing on the graph and nothing in logs.

**Comment by Kroissan** [comment]
image
Here is what should have been bought today :/ (not really optimized i had to say) Anyway idk why the bot didnt buy. Seems it doesnt produce any buy signal in real time mode. Will try in dry run for some time but i guess im doing something wrong and time wont fix it. @drakes00 plz tell me if ur bot does anything.

**Comment by drakes00** [comment]
> image
> Here is what should have been bought today :/ (not really optimized i had to say) Anyway idk why the bot didnt buy. Seems it doesnt produce any buy signal in real time mode. Will try in dry run for some time but i guess im doing something wrong and time wont fix it. @drakes00 plz tell me if ur bot does anything.

Trying to reproduce with verbose.

**Comment by Kroissan** [comment]
`2021-08-20 17:08:21,366 - freqtrade.optimize.backtesting - INFO - Loading data from 2021-08-20 00:00:00 up to 2021-08-20 12:30:00 (0 days).
2021-08-20 17:08:21,366 - freqtrade.optimize.backtesting - INFO - Dataload complete. Calculating indicators
2021-08-20 17:08:21,366 - freqtrade.optimize.backtesting - INFO - Running backtesting for Strategy Dimond
2021-08-20 17:08:21,366 - freqtrade.strategy.interface - DEBUG - Populating indicators for pair FOR/BUSD.
2021-08-20 17:08:21,367 - freqtrade.optimize.backtesting - INFO - Backtesting with data from 2021-08-20 00:00:00 up to 2021-08-20 12:30:00 (0 days).
2021-08-20 17:08:21,368 - freqtrade.strategy.interface - DEBUG - Populating buy signals for pair FOR/BUSD.
2021-08-20 17:08:21,370 - freqtrade.strategy.interface - DEBUG - Populating sell signals for pair FOR/BUSD.
2021-08-20 17:08:21,375 - freqtrade.persistence.models - DEBUG - FOR/BUSD - Assigning new stoploss...
2021-08-20 17:08:21,375 - freqtrade.persistence.models - DEBUG - FOR/BUSD - Stoploss adjusted. current_price=0.10447000, open_rate=0.10447000, max_rate=0.10447000, initial_stop_loss=0.09370959, stop_loss=0.09370959. Trailing stoploss saved us: 0.00000000.
2021-08-20 17:08:21,376 - freqtrade.strategy.interface - DEBUG - FOR/BUSD - Required profit reached. sell_type=SellType.ROI
2021-08-20 17:08:21,376 - freqtrade.persistence.models - DEBUG - FOR/BUSD - Assigning new stoploss...
2021-08-20 17:08:21,376 - freqtrade.persistence.models - DEBUG - FOR/BUSD - Stoploss adjusted. current_price=0.11539000, open_rate=0.11539000, max_rate=0.11800000, initial_stop_loss=0.10350483, stop_loss=0.10350483. Trailing stoploss saved us: 0.00000000.
2021-08-20 17:08:21,376 - freqtrade.strategy.interface - DEBUG - FOR/BUSD - Required profit reached. sell_type=SellType.ROI
2021-08-20 17:08:21,377 - freqtrade.persistence.models - DEBUG - FOR/BUSD - Assigning new stoploss...
2021-08-20 17:08:21,377 - freqtrade.persistence.models - DEBUG - FOR/BUSD - Stoploss adjusted. current_price=0.12001000, open_rate=0.12001000, max_rate=0.12075000, initial_stop_loss=0.10764897, stop_loss=0.10764897. Trailing stoploss saved us: 0.00000000.
2021-08-20 17:08:21,377 - freqtrade.strategy.interface - DEBUG - FOR/BUSD - Required profit reached. sell_type=SellType.ROI`

@drakes00 verbose on what ? backtesting, here it is. When running in real time i just get false for buy and sell on every pairs, i can maybe log that so i can find later if any "true" dropped

**Comment by drakes00** [comment]
Verbose on live trading to see what are the logs when a buy signal appears on the UI's graph I guess

**Comment by mablue** [comment]
> Verbose on live trading to see what are the logs when a buy signal appears on the UI's graph I guess
> 
> EDIT: For whatever reason, it seems that the buy signals are showing way after the actual candle.
> For instance, I have a buy signal on some pair at 3:00PM on UI's graph that was not visible until like 6:00PM and yet no buy neither at 3:00 or 6:00.
> 
> No clue why however.
> Random question, how do the bot handles a negative shift in the frame?
yeh  you find it! :)) :+1: 
https://github.com/freqtrade/freqtrade/issues/5460

**Pull request merged**